packages/frontend/src/features/sqlRunner/store/pieChartSlice.ts

```diff
@@ -35,9 +35,15 @@ export const pieChartConfigSlice = createSlice({
     initialState,
     reducers: {
         setGroupFieldIds: ({ config }, action: PayloadAction<string>) => {
-            if (config?.fieldConfig) {
-                config.fieldConfig.groupFieldIds = [action.payload];
+            if (!config) return;
+
+            if (!config.fieldConfig) {
+                config.fieldConfig = {
+                    groupFieldIds: [action.payload],
+                };
             }
+
+            config.fieldConfig.groupFieldIds = [action.payload];
```

Comment:
Since we don't use a `return` inside the `if` we should use a `else` here.

Reply:
looking!

Reply:
done here: https://github.com/lightdash/lightdash/pull/11009/commits/c153c6618947538ea57037dec7c3f46ccfa5f098